commands/api/endpoint.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package api
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/commands"
+)
+
+func createCommandEndpoint(svc commands.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createCommandReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		cmd := commands.Command{
+			Command:     req.Command,
+			ChannelID:   req.ChannelID,
+			ExecuteTime: req.ExecuteTime,
+		}
+		id, err := svc.CreateCommand(req.token, cmd)
+		if err != nil {
+			return nil, err
+		}
+		res := createCommandRes{
+			ID:      id,
+			created: true,
+		}
+		return res, nil
+	}
+}
+
+func viewCommandEndpoint(svc commands.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(viewCommandReq)
+
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+
+		cmd, err := svc.ViewCommand(req.token, req.id)
+		if err != nil {
+			return nil, err
+		}
+
+		res := viewCommandRes{
+			ID:          cmd.ID,
+			Owner:       cmd.Owner,
+			Name:        cmd.Name,
+			C:           cmd.Command,
+			ChannelID:   cmd.ChannelID,
+			ExecuteTime: cmd.ExecuteTime,
+			Metadata:    cmd.Metadata,
+		}
+		return res, nil
+	}
+}
+
+func listCommandEndpoint(svc commands.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listCommandReq)
+
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+
+		// page, err := svc.ListCommands(ctx)
+		// if err != nil {
+		// 	return nil, err
+		// }
+
+		// res := commandsPageRes{
+		// 	pageRes: pageRes{
+		// 		Total:  page.Total,
+		// 		Offset: page.Offset,
+		// 		Limit:  page.Limit,
+		// 		Order:  page.Order,
+		// 		Dir:    page.Dir,
+		// 	},
+		// 	Commands: []viewCommandRes{},
+		// }
+		// for _, command := range page.Commands {
+		// 	view := viewCommandRes{
+		// 		ID:       command.ID,
+		// 		Metadata: command.Metadata,
+		// 	}
+		// 	res.Commands = append(res.Commands, view)
+		// }
+		// return res, nil
+		return nil, nil
+	}
+}
+
+func updateCommandEndpoint(svc commands.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(updateCommandReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		cmd := commands.Command{
+			ID:       req.id,
+			Name:     req.Name,
```

Comment:
All commands parameters can be updated - Name, Command, ExecutionTime and Metadata.

However, ID cannot be updated.